File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int n = word1.length();
        int m = word2.length();
        int i = 0;
        StringBuilder s = new StringBuilder();
        while(i<n && i<m) {
            s.append(word1.charAt(i));
            s.append(word2.charAt(i));
            i++;
        }
        while(i<n) {
            s.append(word1.charAt(i));
            i++;
        }
        while(i<m) {
            s.append(word2.charAt(i));
            i++;
        }
        return s.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.length();
        int m = word2.length();
        int i = 0;
        string s;
        while(i<n && i<m) {
            s+=word1[i];
            s+=word2[i];
            i++;
        }
        s+=word1.substr(i, n-i);
        s+=word2.substr(i, m-i);
        return s;
    }
};